From 61fa14f77d9cb7b3eb3599cbffb43addb59d44fb Mon Sep 17 00:00:00 2001
From: Adarsh0127-Elite <adarshmoharana027@gmail.com>
Date: Wed, 30 Sep 2026 15:36:24 +0000
Subject: [PATCH 5/9] MiuiCamera: Unlock Leica custom interface and features

- Unlock master Leica customization interface and features (LCC)
- Unlock Leica Color/Vibrant styles (isSupportThemeCV)
- Force Leica camera modes, styles, UI, filters, watermark and sounds
- Exclude physical telephoto overrides (a7, K6, q4) to avoid 2x zoom crash on Duchamp

Signed-off-by: Adarsh0127-Elite <adarshmoharana027@gmail.com>
---
 smali_classes4/Te/b.smali                     | 48 ++-----------------
 ...掗揔掗掓揔掞掟掌掓掙掟揔掾掏掙排掛掗掊.smali |  2 +-
 ...驒騘騙騊騕騟騙驒騟験騑騑験騒驒騿験騑騑験騒.smali | 14 +++---
 3 files changed, 11 insertions(+), 53 deletions(-)

diff --git a/smali_classes4/Te/b.smali b/smali_classes4/Te/b.smali
index 227d8a4d..993c7562 100644
--- a/smali_classes4/Te/b.smali
+++ b/smali_classes4/Te/b.smali
@@ -363,30 +363,11 @@
 .end method
 
 .method public static W()Z
-    .locals 2
-
-    sget-object v0, LVa/b;->u:Ljava/lang/String;
-
-    const-string v1, "LCC"
-
-    invoke-virtual {v0, v1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_0
-
-    sget-boolean v0, LVa/b;->w:Z
-
-    if-nez v0, :cond_0
+    .locals 1
 
     const/4 v0, 0x1
 
     return v0
-
-    :cond_0
-    const/4 v0, 0x0
-
-    return v0
 .end method
 
 .method public static d0()Z
@@ -3542,32 +3523,9 @@
 .method public final i2()Z
     .locals 1
 
-    iget-object p0, p0, LTe/b;->e:L騟験騑驒騑騕驒騘騙騊騕騟騙驒騟験騑騑験騒驒騿験騑騑験騒;
-
-    invoke-virtual {p0}, L騟験騑驒騑騕驒騘騙騊騕騟騙驒騟験騑騑験騒驒騿験騑騑験騒;->S2()Z
-
-    move-result p0
-
-    if-eqz p0, :cond_0
-
-    sget-object p0, LVa/b;->t:Ljava/lang/String;
-
-    const-string v0, "NoMadrid"
-
-    invoke-virtual {p0, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
-
-    move-result p0
-
-    if-nez p0, :cond_0
-
-    const/4 p0, 0x1
-
-    return p0
-
-    :cond_0
-    const/4 p0, 0x0
+    const/4 v0, 0x1
 
-    return p0
+    return v0
 .end method
 
 .method public final j()I
diff --git a/smali_classes4/掙掕掗揔掗掓揔掞掟掌掓掙掟揔掾掏掙排掛掗掊.smali b/smali_classes4/掙掕掗揔掗掓揔掞掟掌掓掙掟揔掾掏掙排掛掗掊.smali
index 6c10fe3e..f48fdfb4 100644
--- a/smali_classes4/掙掕掗揔掗掓揔掞掟掌掓掙掟揔掾掏掙排掛掗掊.smali
+++ b/smali_classes4/掙掕掗揔掗掓揔掞掟掌掓掙掟揔掾掏掙排掛掗掊.smali
@@ -904,7 +904,7 @@
 .method public final a9()Z
     .locals 0
 
-    const/4 p0, 0x0
+    const/4 p0, 0x1
 
     return p0
 .end method
diff --git a/smali_classes4/騟験騑驒騑騕驒騘騙騊騕騟騙驒騟験騑騑験騒驒騿験騑騑験騒.smali b/smali_classes4/騟験騑驒騑騕驒騘騙騊騕騟騙驒騟験騑騑験騒驒騿験騑騑験騒.smali
index c4aebafc..9cc0d778 100644
--- a/smali_classes4/騟験騑驒騑騕驒騘騙騊騕騟騙驒騟験騑騑験騒驒騿験騑騑験騒.smali
+++ b/smali_classes4/騟験騑驒騑騕驒騘騙騊騕騟騙驒騟験騑騑験騒驒騿験騑騑験騒.smali
@@ -141,7 +141,7 @@
 .method public A2()Z
     .locals 0
 
-    instance-of p0, p0, L仃仏仍于仍仉于仄仅他仉仃仅于代仈仅从仆仅从仇;
+    const/4 p0, 0x1
 
     return p0
 .end method
@@ -1243,7 +1243,7 @@
 .method public I4()Z
     .locals 0
 
-    instance-of p0, p0, L仃仏仍于仍仉于仄仅他仉仃仅于代仈仅从仆仅从仇;
+    const/4 p0, 0x1
 
     return p0
 .end method
@@ -2197,7 +2197,7 @@
 .method public S2()Z
     .locals 0
 
-    instance-of p0, p0, L仃仏仍于仍仉于仄仅他仉仃仅于代仈仅从仆仅从仇;
+    const/4 p0, 0x1
 
     return p0
 .end method
@@ -3014,7 +3014,7 @@
 .method public a9()Z
     .locals 0
 
-    instance-of p0, p0, L仃仏仍于仍仉于仄仅他仉仃仅于代仈仅从仆仅从仇;
+    const/4 p0, 0x1
 
     return p0
 .end method
@@ -3341,7 +3341,7 @@
 .method public d9()Z
     .locals 0
 
-    instance-of p0, p0, L仃仏仍于仍仉于仄仅他仉仃仅于代仈仅从仆仅从仇;
+    const/4 p0, 0x1
 
     return p0
 .end method
@@ -4485,7 +4485,7 @@
 .method public p4()Z
     .locals 0
 
-    instance-of p0, p0, L仃仏仍于仍仉于仄仅他仉仃仅于代仈仅从仆仅从仇;
+    const/4 p0, 0x1
 
     return p0
 .end method
@@ -4722,7 +4722,7 @@
 .method public r6()Z
     .locals 0
 
-    instance-of p0, p0, L仃仏仍于仍仉于仄仅他仉仃仅于代仈仅从仆仅从仇;
+    const/4 p0, 0x1
 
     return p0
 .end method
-- 
2.43.0
